controllers/updateservice_controller.go

```diff
@@ -389,6 +391,20 @@ func (r *UpdateServiceReconciler) ensureDeployment(ctx context.Context, reqLogge
 		containers[i].LivenessProbe = original.LivenessProbe
 		containers[i].ReadinessProbe = original.ReadinessProbe
 	}
+	initContainers := updated.Spec.Template.Spec.InitContainers
+	for i := range initContainers {
+		var original *corev1.Container
+		switch initContainers[i].Name {
+		case "graph-data":
+			original = resources.graphDataInitContainer
+		default:
+			reqLogger.Info("encountered unexpected init container in pod", "Container.Name", initContainers[i].Name)
+			continue
+		}
```

Comment:
Looks more wordy than I'd expect, why not simply:

```go
if initContainers[i].Name != `graph-data` {
  logger.Info
  continue
}
original = ...
initContainers[i].Image = original.Image
...
```

Is it likely that we grow more `case`s?

Reply:
I agree. I just cut-and-pasted it and been concentrating on trying to get it to test out but I'll change it.